### DISCUSSION TITLE
terraform apply - tfstate - git push

### DISCUSSION THREAD

**Discussion by pkempec** [dflook/terraform-github-actions#329] [Q&A] [answered]
Hello,
It seems terraform apply does not push current state to github repo.
As a result everytime there is a new change and action is triggered the apply will try to execute it from empty state.

For the first init it works great, but after second push it fails because state was not stored to repository.
Therefore on second trigger it will try to create resources that was already created.

I think I am missing some configuration for the action but I was not able to find it in repo.
I guess I could try to find another action to handle git push but it might be actually tricky because new push would trigger new start of the action and it could result in loop.

How was it designed to avoid duplicate execution?
Is there some flag that would enable github push to specific branch with specific comment?

**Comment by dflook** [comment, marked as answer]
You typically use a [backend](https://developer.hashicorp.com/terraform/language/settings/backends/configuration) to store the state.

**Reply by pkempec** [reply]
Thank you. 
I have literraly just implemented it.
